### PR TITLE
Move "scripts-description" to toplevel

### LINKED ISSUE #6881: Move "scripts-description" to toplevel
With #6672 a new config option `scripts-description` was added to provide description for custom commands.

Since this was implemented as `extra` config option every change here requires a update of the `composer.lock` file. This is not really useful since changes to script descriptions don't affect package states or similar at all.

Thus I propose to move `scripts-description` toplevel which also allows to put them next to `scripts`.

### PR DESCRIPTION
**Pull request by mbrodala**
Fixes #6881

### CODE REVIEW

File: res/composer-schema.json [hunk 1]
```diff
@@ -447,6 +447,10 @@
                 }
             }
         },
+        "scripts-description": {
+            "type": ["object"],
```
Comment (stof) [inline res/composer-schema.json]: would be great to describe it further, by using `additionalProperties` to mark items in it as strings.

File: res/composer-schema.json [hunk 2]
```diff
@@ -447,6 +447,10 @@
                 }
             }
         },
+        "scripts-description": {
```
Comment (stof) [inline res/composer-schema.json]: naming it `script-descriptions` would be more correct, as the plural goes to the last noun in English.

Comment (mbrodala) [reply res/composer-schema.json]: I agree but the base option is called `scripts` so it makes sense to name it after that to keep them closely aligned.

File: res/composer-schema.json (hunk 1 above)
Comment (mbrodala) [reply res/composer-schema.json]: I'll take care of this.

File: res/composer-schema.json [hunk 3]
```diff
@@ -447,6 +447,13 @@
                 }
             }
         },
+        "scripts-descriptions": {
+            "type": ["object"],
+            "description": "Descriptions for scripts listeners, shown in console help.",
```
Comment (stof) [inline res/composer-schema.json]: should be `for script listeners` to be valid English (noun used as qualifiers are always singular)

File: res/composer-schema.json (hunk 2 above)
Comment (stof) [reply res/composer-schema.json]: well, I would still vote for the valid plural form (`scripts` is the valid plural form for `script` too btw)

File: res/composer-schema.json (hunk 3 above)
Comment (mbrodala) [reply res/composer-schema.json]: Fixed for `scripts` now, changed completely for `scripts-descriptions`.

**Comment by Seldaek** [comment]
Thanks

**Pull request merged**